extensions/amp-user-notification/0.1/amp-user-notification.js

```diff
@@ -0,0 +1,389 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Layout} from '../../../src/layout';
+import {assertHttpsUrl} from '../../../src/url';
+import {assert} from '../../../src/asserts';
+import {cidFor} from '../../../src/cid';
+import {getService} from '../../../src/service';
+import {log} from '../../../src/log';
+import {viewerFor} from '../../../src/viewer';
+import {onDocumentReady} from '../../../src/document-state';
+import {xhrFor} from '../../../src/xhr';
+
+
+const TAG_ = 'AmpUserNotification';
+
+/**
+ * @export
+ * @typedef {{
+ *   elementId: string,
+ *   ampUserId: string
+ * }}
+ */
+let PostRequestMetadataDef;
+
+/**
+ * @export
+ * @typedef {{
+ *   showNotification: boolean
+ * }}
+ */
+let PostResponseMetadataDef;
+
+/**
+ * @typedef {{
+ *   promise: !Promise,
+ *   resolve: function(*)
+ * }}
+ */
+let UserNotificationDeferDef;
+
+
+/**
+ * Defines underlying API for Notification components.
+ * @interface
+ */
+class NotificationInterface {
+
+  /**
+   * Promise that is resolved with a boolean on whether this Notification
+   * should be shown or not.
+   * @return {!Promise<boolean>}
+   */
+  shouldShow() {}
+
+  /**
+   * Turns the Notification Component visible
+   * and is resolved with the cid string.
+   * @return {!Promise<string>}
```

Comment:
should looke like this now <img width="541" alt="screen shot 2015-12-28 at 5 03 05 pm" src="https://cloud.githubusercontent.com/assets/354746/12028006/eced03fe-ad84-11e5-8bc4-773c1beb3150.png">